todo/src/Components/TaskActionDialog.tsx

```diff
@@ -0,0 +1,174 @@
+import React from "react";
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogTitle from "@mui/material/DialogTitle";
+import { TodoTaskProps } from "../Todo";
+import { IconButton } from "@mui/material";
+import DeleteIcon from "@mui/icons-material/Delete";
+import EditIcon from "@mui/icons-material/Edit";
+import { Grid } from "@mui/material";
+import { TextField } from "@mui/material";
+import { useForm } from "react-hook-form";
+import { IFormData } from "../Todo";
+import { useState } from "react";
+import { makeStyles } from "@material-ui/styles";
+
+const useStyles = makeStyles({
+  span: {
+    color: "red",
+  },
+});
+
+const TaskActionDialog = ({ task, deleteTask, editTask }: TodoTaskProps) => {
+  const classes = useStyles();
+  const [open, setOpen] = React.useState(false);
+  const [openEdit, setOpenEdit] = React.useState(false);
+  const [name, setName] = useState<string>(task.taskName);
+  const [description, setDescription] = useState<string>(task.description);
```

Comment:
I can see that you are using the useForm library.  So do you really require these 2 useStates for name and description ?

Reply:
In order to maintain state of edited task name and description I'm using two useStates.I think it is required to maintain two separate useStates.

Reply:
Updated and pushed.